### ISSUE TITLE
convert the model of objection

### ISSUE THREAD

**Issue by wangcm12** [NeuromorphicProcessorProject/snn_toolbox#27]
Hi  Bodo,,
Your snn_toolbox is very good. I  am learning SNN. I am learning your code, I want to convert the model of objection, for example ,YOLO, I found your model  is about classification，I tried convert the YOLO (it is Keras  model), but failed.  I read your documentation ,but I  havn't  idea. What should I do?  Should I going to write a function about detecting positioning? Can you give me suggestion?

**Comment by rbodo** [comment]
Hi,

Thanks for your question. Implementing spiking networks for object detection would be a really cool feature for the toolbox. It will need some work though (and I won't be able to do it myself). 

As you say, the toolbox expects classifiers, so whenever the network is evaluated (before and after parsing, and after conversion), the toolbox tries to compute an accuracy score by taking the argmax of the network output and comparing it against the provided labels (in the file `y_test.npz`). This is the part that you will have to change. Instead of providing just one class label per image, you will have to provide (possibly multiple) labels and bounding boxes for each frame. And you will have to tell the toolbox not to compute accuracies, but mean-average-precision scores. 

Another thing that you need to check is that the logging and plotting functions do not break. The toolbox can be configured to write various quantities to disk during and after simulation, and to generate plots of intermediate results etc. A simple workaround would be to just let the toolbox store and plot zeros for the accuracies and other things that you removed. But you may want to add functionality to store and plot your new output so you can debug and evaluate more easily.

I suggest you try to follow the processing pipeline here to see what happens to the output of the network after simulating the SNN:

https://github.com/NeuromorphicProcessorProject/snn_toolbox/blob/5f85df15b3be6496a756609fd50bb5a356014367/snntoolbox/simulation/target_simulators/INI_temporal_mean_rate_target_sim.py#L151

Basically everything after this line is about decoding the network output, logging and plotting results. This is where you will have to modify most.

Here is one of the places where the ANN get's evaluated, which will have to be modified as well:

https://github.com/NeuromorphicProcessorProject/snn_toolbox/blob/5f85df15b3be6496a756609fd50bb5a356014367/snntoolbox/parsing/model_libs/keras_input_lib.py#L165

Good luck!

**Comment by wangcm12** [comment]
Thanks!
I found the h5 file of vgg not in  folder. I used the model trained by my own data set， but it reported a error"can not found the model",  I edit iconfig  according to the given example，any suggestion?

**Comment by rbodo** [comment]
The h5 file of VGG is too big to put in the repository.

You need to set the parameter `filename_ann` to the name of your model (without the `.h5`). In addition, you need to tell the toolbox where to find the model:

If you **don't** set the parameter `path_wd` in your config file, then it will be set to the folder where your config file is. In that case, you need to put your model into the folder where the config file is.

Otherwise, just set the `path_wd` to where your model is.

**Comment by wangcm12** [comment]
I did what you said. ,the parameter path_wd was set, the model in the folder where the config file is. but it  reported "  No model  found in config file ”，I spent a day finding the problem, but still not solved. Help!

**Comment by rbodo** [comment]
Please send me the full error trace, your config file, and a picture of your file structure (where the model and config file are).

**Comment by wangcm12** [comment]
sorry，I reply by QQ E-mail, the  file upload failed.  I  tried convert the caffemodel,it succeeded，the config file is ok. I think I probably know the reason of the error. When I ran my  Python file on my computer, it still report  the same error as this"NO model in the config file", so it should be my Python or keras problem.
